src/lib/Microsoft.Health.Events/EventHubProcessor/AssignedPartitionProcessor.cs

```diff
@@ -0,0 +1,226 @@
+            TokenCredential tokenCredential,
+            string eventHubName,
+            string fullyQualifiedNamespace,
+            EventProcessorOptions clientOptions = default)
+                : base(
+                    new BlobCheckpointStore(checkpointClient.GetBlobContainerClient()),
+                    eventBatchMaximumCount,
+                    consumerGroup,
+                    fullyQualifiedNamespace,
+                    eventHubName,
+                    tokenCredential,
+                    clientOptions)
+        {
+            _logger = logger;
+            _eventConsumerService = eventConsumerService;
+            _checkpointClient = checkpointClient;
+            _assignedPartitions = assignedPartitions
+                ?? throw new ArgumentNullException(nameof(assignedPartitions));
+        }
+
+        // To simplify logic, tell the processor that only its assigned
+        // partitions exist for the Event Hub.
+
+        protected override Task<string[]> ListPartitionIdsAsync(
+            EventHubConnection connection,
+            CancellationToken cancellationToken) =>
+                Task.FromResult(_assignedPartitions);
+
+        // Tell the processor that it owns all of the available partitions for the Event Hub.
+        protected override Task<IEnumerable<EventProcessorPartitionOwnership>> ListOwnershipAsync(
+            CancellationToken cancellationToken) =>
+                Task.FromResult(
+                    _assignedPartitions.Select(partition =>
+                        new EventProcessorPartitionOwnership
+                        {
+                            FullyQualifiedNamespace = FullyQualifiedNamespace,
+                            EventHubName = EventHubName,
+                            ConsumerGroup = ConsumerGroup,
+                            PartitionId = partition.ToString(),
+                            OwnerIdentifier = Identifier,
+                            LastModifiedTime = DateTimeOffset.UtcNow,
+                        }));
+
+        // Accept any ownership claims attempted by the processor; this allows the processor to
+        // simulate renewing ownership so that it continues to own all of its assigned partitions.
+        protected override Task<IEnumerable<EventProcessorPartitionOwnership>> ClaimOwnershipAsync(
+           IEnumerable<EventProcessorPartitionOwnership> desiredOwnership,
+           CancellationToken cancellationToken) =>
+                Task.FromResult(desiredOwnership.Select(ownership =>
+                {
+                    ownership.LastModifiedTime = DateTimeOffset.UtcNow;
+                    return ownership;
+                }));
+
+        protected async override Task OnProcessingEventBatchAsync(
+            IEnumerable<EventData> events,
+            EventProcessorPartition partition,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (events.Count() == 0)
```

Comment:
thanks Dustin, that makes sense. I updated the code. Instead of using instead of events.Count() the code now uses a bool which will get unset if there are actually events, and if there are not events then it is a wait event (empty event) which we can pass through to the batching service. Yes the timer triggers an empty event which is passed into this method.